lib/client.js

```diff
@@ -52,78 +54,26 @@ function handleJSONResponse(request, bigint, callback) {
     response.on('end', function () {
       if (response.statusCode < 200 || response.statusCode > 299) {
         err = new SolrError(request, response, text);
-        if (callback) callback(err, null);
+        if (callback) {
+          callback(err, null);
+        }
       } else {
         try {
           data = pickJSON(bigint).parse(text);
         } catch (error) {
           err = error;
         } finally {
-          if (callback) callback(err, data);
+          if (callback) {
+            callback(err, data);
+          }
         }
       }
     });
   };
 }
 
-/**
- * HTTP POST request. Send update commands to the Solr server (commit, add, delete, optimize)
- *
- * @param {Object} params
- * @param {String} params.host - IP address or host address of the Solr server
- * @param {Number|String} params.port - port of the Solr server
- * @param {String} params.core - name of the Solr core requested
- * @param {String} params.authorization - value of the authorization header
- * @param {String} params.fullPath - full path of the request
- * @param {String} params.json -
- * @param {Boolean} params.secure -
- * @param {Boolean} params.bigint -
- * @param {http.Agent} [params.agent] -
- * @param {Function} callback(err,obj) - a function executed when the Solr server responds or an error occurs
- * @param {Error} callback().err
- * @param {Object} callback().obj - JSON response sent by the Solr server deserialized
- *
- * @return {http.ClientRequest}
- * @api private
- */ function postJSON(params, callback) {
```

Comment:
was this not used anywhere?

Reply:
It was, but it was a very large chunk of duplicated code. It is all contained in `doRequest` now. I know you said you do not like the idea of a generic request function, but it's very easy to split again later. The GET and POST requests only differed by one or two lines, which are neatly contained in `if (method === "POST")` now. There are no other methods in use yet in this library. If the requirements diverge later, we can simply split up the `doRequest` method.

Reply:
Sure, let's do it that way then